packages/salesforcedx-vscode-apex/src/telemetry/index.ts

```diff
@@ -4,5 +4,5 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import { TelemetryService } from '@salesforce/salesforcedx-utils-vscode';
-export const telemetryService = TelemetryService.getInstance();
+import { TelemetryProvider } from '@salesforce/salesforcedx-utils-vscode';
+export const telemetryService = TelemetryProvider.getInstance();
```

Comment:
Instead of changing to export the Provider, could we put an optional param on the call to TelemetryService.getInstance(); ?  Like  TelemetryService.getInstance(context?: ExtensionContext);

Then internally the telemetry service getInstance method can use the Provider to return the correct instance. If we can make this work, then it seems like it could be backwards compatible and we can just change the way we make the call to send the context once the changes live in the core extension.

Reply:
Thanks @klewis-sfdc , brilliant suggestion! I just made changes based on your idea.